### PR TITLE
Feat: tests table

### LINKED ISSUE #147: execution list [tests]: table: Path, Status (6 circles, one for each existing statuses); each execution item is a first level grouping of tests (e.g. alsa); show status for all tests in that grouping; by clicking on a grouping the accordion expands (shows at most 10 items in a scrollable list), each item is an individual test, by pressing an individual test the Test Details Page opens.

### PR DESCRIPTION
**Pull request by mari1912**
- closes #147

- WIP: adjust the click in the entire row

image

### CODE REVIEW

File: backend/kernelCI_app/views/groupedTestsView.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    def get(self, request):
+        commit = request.GET.get('git_commit_hash')
+        branch = request.GET.get('git_repository_branch')
+        url = request.GET.get('git_repository_url')
+        origin = request.GET.get('origin')
+        if not commit:
+            return HttpResponseBadRequest(
+                getErrorResponseBody('missing required field `git_commit_hash`')
+            )
```
Comment (lfjnascimento) [inline backend/kernelCI_app/views/groupedTestsView.py:22]: we can do something like

```
    get_required_params = ['git_commit_hash', 'git_repository_branch', 'git_repository_url' 'origin']

    def validate_required_params(self, request):
        for p in self.get_required_params:
            if not request.GET.get(p):
                return HttpResponseBadRequest(
                getErrorResponseBody(f'missing required field `{p}`')
            )
```

we can even put `validate_required_params` in utils file so we can reuse it as `validate_required_params(params, request)`

File: backend/kernelCI_app/views/groupedTestsView.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            Q(build__checkout__origin=origin)
+
+        if path:
+            if not path.endswith('.'):
+                path += '.'
+            filterQ &= Q(path__startswith=path)
+
+        result = (
+            Tests.objects
+            .filter(filterQ)
```
Comment (lfjnascimento) [inline backend/kernelCI_app/views/groupedTestsView.py:50]: we can just:

```suggestion
        if path:
            if not path.endswith('.'):
                path += '.'

        result = (
            Tests.objects
            .filter(
                build__checkout__git_commit_hash=commit, 
                build__checkout__git_repository_branch=branch, 
                build__checkout__git_repository_url=url, 
                build__checkout__origin=origin,
                path__startswith=path,
            )
```

File: backend/kernelCI_app/views/groupedTestsView.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            .annotate(
+                fail_tests=Sum(Case(When(status='FAIL', then=1), default=0)),
+                error_tests=Sum(Case(When(status='ERROR', then=1), default=0)),
+                miss_tests=Sum(Case(When(status='MISS', then=1), default=0)),
+                pass_tests=Sum(Case(When(status='PASS', then=1), default=0)),
+                done_tests=Sum(Case(When(status='DONE', then=1), default=0)),
+                skip_tests=Sum(Case(When(status='SKIP', then=1), default=0)),
+                null_tests=Sum(Case(When(status__isnull=True, then=1), default=0)),
+                total_tests=Count('id'),
+                path_group=F('first_path_segment'),
```
Comment (lfjnascimento) [inline backend/kernelCI_app/views/groupedTestsView.py:70]: this doesn't seem to be being used

File: backend/kernelCI_app/views/groupedTestsView.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        )
+
+        for item in result:
+            item['individual_tests'] = [
+                {
+                    'path': test.split(', ')[0],
+                    'status': test.split(', ')[1],
+                    'start_time': test.split(', ')[2],
+                    'duration': test.split(', ')[3]
+                }
```
Comment (lfjnascimento) [inline backend/kernelCI_app/views/groupedTestsView.py:92]: create the splited array just one time and reuse it
```
 fileds = test.split(', ')
 {
'path': fields[0],
 'status': fields[1],
 ...
```

File: backend/kernelCI_app/urls.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
 urlpatterns = [
     path("tree/", views.TreeView.as_view(), name="tree"),
     path("tree/<str:commit_hash>/commits", views.TreeCommitsHistory.as_view(), name="treeCommits"),
-    path("tree/tests/", views.revisionTests.as_view(), name="treeRevisionTests"),
+    path("tree/tests/", views.groupedTests.as_view(), name="treeGroupedTests"),
```
Comment (WilsonNet) [inline backend/kernelCI_app/urls.py:8]: Can you add a httpie request for it in the requests folder?

File: backend/kernelCI_app/views/groupedTestsView.py (hunk 3 above)
Comment (mari1912) [reply backend/kernelCI_app/views/groupedTestsView.py:70]: what? path_group? it is

**Comment by WilsonNet** [comment]
image
Arrow should look down when it is opened (don't need animation) (just do a transform rotateZ)

File: dashboard/src/components/Accordion/Accordion.tsx [hunk 6]
```diff
@@ -141,29 +160,68 @@ const AccordionBuildsTrigger = ({
...
+      <TableCell>{triggerInfo.path_group}</TableCell>
       <TableCell className="flex flex-row gap-1">
-        <ColoredCircle
-          quantity={triggerInfo.testErrors ?? 0}
-          backgroundClassName={ItemType.Error}
-        />
-        <ColoredCircle
-          quantity={triggerInfo.testSuccessfull ?? 0}
-          backgroundClassName={ItemType.Warning}
+        <TestStatus
```
Comment (WilsonNet) [inline dashboard/src/components/Accordion/Accordion.tsx:168]: add a cursor: pointer here and a hover that shows the user that it is a distinc click (maybe a filter brightnesss (0.6) )

File: dashboard/src/components/Accordion/Accordion.tsx [hunk 7]
```diff
@@ -141,29 +160,68 @@ const AccordionBuildsTrigger = ({
...
+          search: s => s,
+        });
+      }
+    },
+    [navigate],
+  );
+
+  const rows = useMemo(() => {
+    return data.map(test => (
+      <TableRow onClick={onClickName} key={test.path}>
```
Comment (WilsonNet) [inline dashboard/src/components/Accordion/Accordion.tsx:208]: if it is clickable it should haver a `cursor: pointer`

File: dashboard/src/components/Accordion/Accordion.tsx [hunk 8]
```diff
@@ -16,29 +20,28 @@ import {
 import ColoredCircle from '../ColoredCircle/ColoredCircle';
 import { ItemType } from '../ListingItem/ListingItem';
 
+import { TestStatus } from '../Status/Status';
```
Comment (WilsonNet) [inline dashboard/src/components/Accordion/Accordion.tsx:23]: nit: please avoid import with `../`

File: backend/kernelCI_app/views/groupedTestsView.py (hunk 3 above)
Comment (lfjnascimento) [reply backend/kernelCI_app/views/groupedTestsView.py:70]: ok, my bad

**Comment by mari1912** [comment]
> image Arrow should look down when it is opened (don't need animation) (just do a transform rotateZ)

i will create a task to register this, it is taking me too much time right now, since it is necessary to deal with the collapsible state from radix ui or we will need to control it by ourself

**Review by WilsonNet — APPROVED** [review]
working nice

**Pull request merged**